core/species.cpp

```diff
@@ -4161,8 +4699,29 @@ void Species::RecordNewHaplosome(std::vector<slim_position_t> *p_breakpoints, Ha
 	
 	right = (double)chromosome.last_position_+1;
 	tsk_id_t parent = (tsk_id_t) (polarity ? haplosome1TSKID : haplosome2TSKID);
-	int ret = tsk_edge_table_add_row(&tables_.edges, left, right, parent, offspringTSKID, NULL, 0);
+	int ret = tsk_edge_table_add_row(&tsinfo.tables_.edges, left, right, parent, offspringTSKID, NULL, 0);
 	if (ret < 0) handle_error("tsk_edge_table_add_row", ret);
+	
+	// BCH 12/10/2024: With the new metadata scheme for haplosome, we also need to fix the is_null_metadata if
+	// the new haplosome is a null haplosome *and* it belongs to a chromosome type where that is notable.  In
+	// the present design, that can only be chromosome types "A" and "H"; the other chromosome types do not
+	// allow deviation from the default null-haplosme configuration.
```

Comment:
IIUC - the node `is_null` metadata not record that, for instance, the second `X` of a male is null, since that goes without saying?